Userland/DevTools/HackStudio/HackStudioWidget.cpp

```diff
@@ -611,6 +614,37 @@ NonnullRefPtr<GUI::Action> HackStudioWidget::create_show_in_file_manager_action(
     return show_in_file_manager_action;
 }
 
+NonnullRefPtr<GUI::Action> HackStudioWidget::create_copy_relative_path_action()
+{
+    auto copy_relative_path_action = GUI::Action::create("Copy &Relative Path", [this](const GUI::Action&) {
+        auto paths = selected_file_paths();
+        VERIFY(!paths.is_empty());
+        auto paths_string = String::join("\n", paths);
+        GUI::Clipboard::the().set_plain_text(paths_string);
+    });
+    copy_relative_path_action->set_enabled(true);
+    copy_relative_path_action->set_icon(GUI::Icon::default_icon("hard-disk").bitmap_for_size(16));
+
+    return copy_relative_path_action;
+}
+
+NonnullRefPtr<GUI::Action> HackStudioWidget::create_copy_full_path_action()
+{
+    auto copy_full_path_action = GUI::Action::create("Copy &Full Path", [this](const GUI::Action&) {
+        auto paths = selected_file_paths();
+        VERIFY(!paths.is_empty());
+        Vector<String> full_paths;
+        for (auto& path : paths)
+            full_paths.append(get_absolute_path(path));
+        auto paths_string = String::join("\n", full_paths);
```

Comment:
Rather than build a temporary Vector, you could directly append the paths to a StringBuilder. Something like this:

```suggestion
        StringBuilder builder;
        for (auto& path : paths) {
            builder.append(get_absolute_path(path));
            builder.append("\n");
        }
        auto paths_string = builder.to_string();
```

Reply:
You're completely right about the newlines! And I'm micro-optimising something that isn't called often anyway. :sweat_smile: